tbx/project_styleguide/templates/patterns/pages/work/historical_work_page.html

```diff
@@ -1,89 +1,20 @@
-{% extends "patterns/base_page.html" %}
+{% extends "patterns/pages/work/work_page_base.html" %}
 {% load wagtailcore_tags wagtailimages_tags static %}
 
-{% block meta_tags %}
-    <script>
-        window.dataLayer = window.dataLayer || [];
-        window.dataLayer.push({
-            'author-title' : '{{ page.first_author.role }}',
-            'author-name' : '{{ page.first_author.name }}',
-            'published' : '{{ page.first_published_at|date:"Ymd" }}',
-            'tags' : [{% for tag in page.related_services.all %}'{{tag.name}}'{% if not forloop.last %}, {% endif %}{% endfor %}],
-            'read-time' : '{{ page.read_time }}',
-            'id': '{{ page.id }}',
-            'page-h1' : '{{ page.title }}',
-            'page-title' : '{{ page.title }} | Torchbox',
-            'page-path': '{{ page.slug }}',
-            'event': 'page-information'
-        });
-    </script>
-    <meta name="author" content="{{ page.first_author.name }}">
-    <meta property="og:article:published_time" content="{{ page.date|date:'Y-m-d' }}" />
-    {{ block.super }}
-
+{# The historical work pages did not have a separate intro - so we just loop over the body streamfield to find the intro block(s) #}
+{% block intro %}
+    {% for intro_block in page.body %}
+        {% if intro_block.block_type == 'intro' %}
+            {% include_block intro_block %}
+        {% endif %}
+    {% endfor %}
```

Comment:
**Thoughts (non-blocking)**:

1. If there are multiple intros (which is highly unlikely), they will be rendered.
2. Instead of looping through all the blocks, I wonder whether it would be more efficient to have a (cached) property on the model, where we just extract the first intro block from the body StreamField, then use it here.

Reply:
Thanks @engineervix - I have updated to get the first intro as a cached property. But, this now means that if there were more than one intro, the second would never be rendered. Is there an easy way to add another cached property that returns the body minus the first intro?

Reply:
regarding the more than one intro problem, I checked existing HistoricalWorkPages:

```console
>>> pages = HistoricalWorkPage.objects.all()
>>> [sum(1 for block in page.body if block.block_type == "intro") for page in pages]
[1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
```

Looks like we only have 1 page with more than 1 intro:

```console
>>> with_multiple_intros = [page for page in pages if sum(1 for block in page.body if block.block_type == "intro") > 1]
>>> with_multiple_intros
[<HistoricalWorkPage: Increasing Samaritans online donations revenue by 200%>]
>>> with_multiple_intros[0].live
True
>>> 
```

So maybe we can have the second intro for that page edited manually (either combining it with first intro, if it makes sense to do so, or maybe use another suitable block type, like paragraph)

Reply:
In this case, then I think we won't need to add another property to return the body content without first intro